elements/lisk-codec/src/codec.ts

```diff
@@ -113,6 +120,39 @@ export class Codec {
 		return (res as unknown) as T;
 	}
 
+	// For performance applications use decode() instead!
+	public decodeJSON<T>(schema: Schema, message: Buffer): T {
+		const decodedMessage: IteratableGenericObject = this.decode(
+			schema,
+			message,
+		);
+		const decodedMessageCopy = objectUtils.cloneDeep(decodedMessage);
+		decodedMessageCopy[Symbol.iterator] = iterator;
+
+		recursiveTypeCast(
+			'toJSON',
+			decodedMessageCopy,
+			(schema as unknown) as SchemaProps,
```

Comment:
Why not pass `scehma` as it is, and treat it as you want inside the function?

Reply:
Sorry @nazarhussain I do not understand this question could you rephrase/expand please?

Reply:
You are typecasting `schema` to `SchemaProps`, looks like both are same otherwise would not work.  So why not pass the schema without `as SchemaProps`

Reply:
Let's leave as is. Maybe let's discuss complete type refactor for lisk-codec if deemed necessary as this abstract structures structures might need better type abastractions.